examples/seeded_extractor.rs

```diff
@@ -0,0 +1,19 @@
+#![feature(async_await, futures_api)]
+
+use tide::head::{NamedHeader, Header, SegmentName, Named};
+use http::header::{HeaderName, HeaderValue};
+
+async fn display_header(value: Header<HeaderValue>) -> String {
+    String::from_utf8_lossy(value.0.as_ref()).into_owned()
+}
+
+async fn display_number(nr: Named<i32>) -> String {
+    format!("Segment number: {}", nr.0)
+}
+
+fn main() {
+    let mut app = tide::App::new(());
+    app.at("/").get_with(display_header, NamedHeader(HeaderName::from_static("user-agent")));
```

Comment:
No, rather I missed something because in the macro expansion for one argument, this `let` binding gets created: `let ($($X),)* = …`. Which of course then expands to `let (T0) = ` and not a tuple as in `let (T0,) = …`. And the same for the right side of that assignment.

I must however say it felt more ergonomic to write, it is much less confusing with one less pair of brackets and comma.